system/jlib/jlog.hpp

```diff
@@ -1343,4 +1345,345 @@ interface IComponentLogFileCreator : extends IInterface
+    else
+        throw makeStringExceptionV(-1, "Encountered unknown Log Access Format name: '%s'", name);
+}
+
+interface IRemoteLogAccess : extends IInterface
+{
+    virtual bool fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format) = 0;
+
+    virtual const char * getRemoteLogAccessType() const = 0;
+    virtual IPropertyTree * queryLogMap() const = 0;
+    virtual const char * fetchConnectionStr() const = 0;
+};
+
+inline bool fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, const StringArray & cols, LogAccessLogFormat format)
+{
+    LogAccessConditions logFetchOptions;
+    logFetchOptions.setTimeRange(timeRange);
+    logFetchOptions.setFilter(filter);
+    logFetchOptions.copyLogFieldNames(cols); //ensure these fields are declared in m_logMapping->queryProp("WorkUnits/@contentcolumn")? or in LogMap/Fields?"
+
+    return logAccess.fetchLog(logFetchOptions, returnbuf, format);
+}
+
+//logAccessPluginConfig expected to contain connectivity and log mapping information
+typedef IRemoteLogAccess * (*newLogAccessPluginMethod_t_)(IPropertyTree & logAccessPluginConfig);
+
+class jlib_decl RemoteLogAccessLoader
+{
+public:
+    RemoteLogAccessLoader() = default;
+    ~RemoteLogAccessLoader();
+
+    template <class RemoteLogAccessPlugin>
+    static RemoteLogAccessPlugin * queryRemoteLogAccessor()
+    {
+        Owned<IPropertyTree> logAccessPluginConfig = getGlobalConfigSP()->getPropTree("logAccess");
+#ifdef LOGACCESSDEBUG
+        if (!logAccessPluginConfig)
+        {
+            const char *simulatedGlobalYaml = R"!!(global:
+              logAccess:
+                name: "localES"
+                type: "elasticstack"
+                connection:
+                  protocol: "http"
+                  host: "localhost"
+                  port: 9200
+                logMaps:
+                - type: "global"
+                  storeName: "filebeat-7.9.3-*"
+                  searchColumn: "message"
+                  timeStampColumn: "created_ts"
+                - type: "workunits"
+                  storeName: "filebeat-7.9.3-*"
+                  searchColumn: "hpcc.log.jobid"
+                - type: "components"
+                  searchColumn: "kubernetes.container.name"
+                - type: "audience"
+                  searchColumn: "hpcc.log.audience"
+                - type: "class"
+                  searchColumn: "hpcc.log.class"
+            )!!";
+            Owned<IPropertyTree> testTree = createPTreeFromYAMLString(simulatedGlobalYaml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
+            logAccessPluginConfig.setown(testTree->getPropTree("global/logAccess"));
+        }
+#endif
+        if (!logAccessPluginConfig)
+            throw makeStringException(-1, "RemoteLogAccessLoader: logaccess configuration not available!");
+
+        static constexpr const char * methodName = "loadLogAccessPlugin";
```

Comment:
note: I've just discovered that __func__ is defined by the c++11 standard as the name of the function.  It would be cleaner to use that.  I may start using it myself...